### DISCUSSION TITLE
Next.js - Two apps, one for Blog, on /blog of domain - SEO okay?

### DISCUSSION THREAD

**Discussion by Hewlbern** [vercel/next.js#17689] [Help] [answered]
Hi everyone!

I've just finished creating my site at https://moodmap.app . 

I would like to build out https://strapi.io/blog/build-a-blog-with-next-react-js-strapi-and-apollo this for the blog section, but don't want to add apollo to my current site, as it seems like it would be annoying - currently using react-query. Also my main site I have some load speed issues to fix and don't want to complicate things when i can just use this tutorial site above.

If i use the new feature from next to use a different base url, being /blog, would the SEO flow through to the main index site?

Alternatively - would using react-query be difficult to implement the same functionality?

Thanks!

**Comment by leerob** [comment, marked as answer]
Apollo is very much overkill for what that tutorial is trying to accomplish. You can use `graphql-request` with `getStaticProps`/`getStaticPaths` 👍  This prevents all of the Apollo setup needed.

This code snippet is for Hasura, but the idea is the same. 

```js
// lib/api.js

import { GraphQLClient } from 'graphql-request';

const getTodos = async () => {
  const graphQLClient = new GraphQLClient(process.env.HASURA_API_URL, {
    headers: {
      'x-hasura-admin-secret': process.env.HASURA_ADMIN_SECRET
    }
  });

  const query = `
    {
      todos {
        id
        createdAt
        isCompleted
        title
      }
    }
  `;

  try {
    return graphQLClient.request(query);
  } catch (error) {
    return { error };
  }
};

export { getTodos };
```
https://github.com/prisma-labs/graphql-request

**Reply by Hewlbern** [reply]
Thanks for this!

Just trying to implement it now,

```
import { GraphQLClient } from 'graphql-request';

const GRAPHQL_URL = process.env.API_URL || "http://localhost:1337";


const getTodos = async () => {
  const graphQLClient = new GraphQLClient(GRAPHQL_URL, {
    headers: {
      'x-graphql-admin-secret': GRAPHQL_URL
    }
  });

  const query = `
    {
      todos {
        id
        createdAt
        isCompleted
        title
      }
    }
  `;

  try {
    return graphQLClient.request(query);
  } catch (error) {
    return { error };
  }
};

export { getTodos };
```

Should this be okay?
